extensions/ql-vscode/src/authentication.ts

```diff
@@ -41,22 +49,33 @@ export class Credentials {
     // Sessions are changed when a user logs in or logs out.
     context.subscriptions.push(vscode.authentication.onDidChangeSessions(async e => {
       if (e.provider.id === GITHUB_AUTH_PROVIDER_ID) {
-        this.octokit = await this.createOctokit(false);
+        this.octokitAndToken = await this.createOctokit(false);
       }
     }));
   }
 
   async getOctokit(): Promise<Octokit.Octokit> {
-    if (this.octokit) {
-      return this.octokit;
+    if (this.octokitAndToken) {
+      return this.octokitAndToken.octokit;
     }
 
-    this.octokit = await this.createOctokit(true);
+    this.octokitAndToken = await this.createOctokit(true);
     // octokit shouldn't be undefined, since we've set "createIfNone: true".
     // The following block is mainly here to prevent a compiler error.
-    if (!this.octokit) {
+    if (!this.octokitAndToken) {
+      throw new Error('Did not initialize Octokit.');
+    }
+    return this.octokitAndToken.octokit;
+  }
+
+  async getToken(): Promise<string> {
+    if (this.octokitAndToken) {
+      return this.octokitAndToken.token;
+    }
+    this.octokitAndToken = await this.createOctokit(true);
+    if (!this.octokitAndToken) {
       throw new Error('Did not initialize Octokit.');
     }
-    return this.octokit;
+    return this.octokitAndToken.token;
   }
```

Comment:
Minor: You have a neat "lazy initialisation" pattern here, and you can make it even simpler to avoid the repetition between the `getOctokit` and `getToken` functions.

Couple of ideas:
- Is it possible to get the token string back out of an `Octokit` object? Then you can avoid the `OctokitAndToken` type with its separate `token` field. `await octokit.auth()` might do it, but I haven't tried! Otherwise stick with what you have.
- Consider having a private helper function here `getOctokitAndToken()` which does most of the work, including the creation and error handling, and returns `this.octokitAndToken`. Then `getOctokit()` becomes `return (await this.getOctokitAndToken()).octokit` (and `getToken` is similar).

Reply:
Thank you for pointing out `octokit.auth()`. I didn't think there was a way to get the token back out of the octokit object. But I've tried this out locally and it works perfectly, so I agree we can remove all these changes to `authentication.ts` and instead use `.auth()` when we need the token.